ParselyTracker/ParselyTracker.swift

```diff
@@ -66,12 +72,25 @@ public class Parsely {
      - Parameter extraData: A dictionary of additional information to send with the generated pageview event
      - Parameter siteId: The Parsely site ID for which the pageview event should be counted
      */
-    public func trackPageView(url: String,
-                              urlref: String = "",
-                              metadata: ParselyMetadata? = nil,
-                              extraData: Dictionary<String, Any>? = nil,
-                              siteId: String = "")
-    {
+    public func trackPageView(
+        url: String,
+        urlref: String = "",
+        metadata: ParselyMetadata? = nil,
+        extraData: Dictionary<String, Any>? = nil,
+        siteId: String = ""
+    ) {
+        eventProcessor.async {
+            self._trackPageView(url: url, urlref: urlref, metadata: metadata, extraData: extraData, siteId: siteId)
+        }
+    }
+
+    private func _trackPageView(
```

Comment:
Have you considered keeping the code inside `eventProcessor.async { }`?

Nitpick. This approach makes the body of the private version clean but... I don't know... I feel like all these parameters redefined just for one call make the code heavier 🤷‍♂️ 😅 

Or is there a runtime/performance reason for this?

No worries either way.

Reply:
I wanted to make this PR easier to review. So, I avoided making too many changes to the existing code. I can follow up you suggestions in a separate PR though. What do you think?